server/src/test/java/org/opensearch/common/geo/PointParserTests.java

```diff
@@ -0,0 +1,216 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.common.geo;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.geometry.Point;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public class PointParserTests extends OpenSearchTestCase {
+    private static final PointParser PARSER;
+    private static final PointParser PARSER_WITHOUT_GEOHASH;
+    private static final String FIELD_NAME_X = "x";
+    private static final String FIELD_NAME_Y = "y";
+    private static final String FN_GEOHASH = "geohash";
+    private static final String FN_GEOJSON_TYPE = "type";
+    private static final String FV_GEOJSON_TYPE_POINT = "Point";
+    private static final String FN_GEOJSON_COORDS = "coordinates";
+
+    private static final String FN_IN_ERR_MSG = "test error message";
+
+    static {
+        PARSER = new PointParser(FIELD_NAME_X, FIELD_NAME_Y, true);
+        PARSER_WITHOUT_GEOHASH = new PointParser(FIELD_NAME_X, FIELD_NAME_Y, false);
+    }
+
+    public void testParseFromObjectOfBasicForm() throws IOException {
+        double x = 74.00;
+        double y = 40.71;
+        XContentBuilder pointInJson = XContentFactory.jsonBuilder().startObject().field(FIELD_NAME_X, x).field(FIELD_NAME_Y, y).endObject();
+        try (XContentParser parser = createParser(pointInJson)) {
+            parser.nextToken();
+            Point point = PARSER.parseObject(parser, true, GeoUtils.EffectivePoint.BOTTOM_LEFT);
+            assertEquals(x, point.getX(), 0.001);
+            assertEquals(y, point.getY(), 0.001);
```

Comment:
there is a precision value defined, try to use that. 0.001 is a magic number.

https://github.com/opensearch-project/OpenSearch/blob/996d33adb22ac6a523962166f3677f8dc6ac9c1f/server/src/test/java/org/opensearch/index/search/geo/GeoPointParsingTests.java#L52

Reply:
ACK